### DISCUSSION TITLE
Avoiding barriers to contributing

### DISCUSSION THREAD

**Discussion by OscartGiles** [alan-turing-institute/REG-handbook#10] [Ideas]
### Possible Barriers

1. Having to set up a local environment to make changes.
    - But @JimMadge pointed out you can click `edit` page and then commit changes from GitHib markdown editor 👌 
2. Having to submit a PR that is reviewed, and potentially being asked to make more changes. If you can't make a minor edit and forget about it you might not bother (you've already got a busy day ahead).
    - What if we allow commits straight to `main`. Run tests to check everything builds and then deploy to master. Then optionally ask someone to review after release if necessary. You can then roll back changes if they are not acceptable.

**Comment by OscartGiles** [comment]
I suspect information on the wiki is more likely to be out of date/wrong, compared to the risk that someone will put wrong (but new) information on there. This makes me think we should minimise the effort to make changes as far as possible.

**Comment by JimMadge** [comment]
I think keeping contribution easy is going to be key to having a successful book.

I was concious about trying to keep development simple. I think the requirements [are small](https://alan-turing-institute.github.io/REG-handbook/docs/contributing/getting_started/) and give people the option to use their preferred text editors and git clients. One of the advantages of going with Hugo over Jekyll is that it is distributed as a binary so you don't need to manage Ruby packages.

I did [mention](https://alan-turing-institute.github.io/REG-handbook/docs/contributing/editing_a_page/) the links to edit pages in browser, but maybe that is worth promoting more if people like making edits that way.

What I have tried to do in the contribution guide is to put more of the burden on [the reviewers](https://alan-turing-institute.github.io/REG-handbook/docs/contributing/reviewing_changes/) (and hopefully most on the CI). My idea is that the reviewers are going to be more dedicated to the book, so they can take the time to review changes. Maybe they could even act as editors so that contributors are not asked to make trivial or style changes to their PRs?

When it comes to making minor edits. I hope it should work out naturally. Small edits require less scrutiny from the reviewers and are less likely to require significant (or any) changes.

All of this is up for discussion and open to change though. Especially if we find the process to be blocking contributions. A fast track for small or urgent edits could be a good idea. Maybe we could have a tag for this.

**Comment by andrewphilipsmith** [comment]
A quick comment from me. I've attempted to make two changes (#15 and #14) and both times I've fallen foul of the markdown linter. I think that some of the lining rules should be relaxed. As presently configured I think they risk being a barrier to contributing.

* The changes in both the PRs render fine in a local hugo instance, in VSC markdown preview and Joplin's markdown preview. 
* When editing in the browser you don't get any real-time feedback about the lining rules.
* Very subjective, but I think that some of the rules (dashes for bullet points, no double line spaces) make the raw markdown less easy to read.

I will look at the specifics in due course, but I wanted to add a note whilst it is fresh in my mind.

**Reply by JimMadge** [reply]
Hi @andrewphilipsmith thanks for the feedback.

I think we should have a consensus before changing anything, which is difficult at the moment. The currently linting rules definitely follow my personal preferences.

That said, I think ensuring a consistent style across a collaborative code base is very important. I would worry that loosening the rules would lead to inconsistent styles across sections of the handbook making it much less readable and editable in the long run. At the risk of being flippant, I would say that the CI failing is a sign that it is working as intended.

With regards to rendering there are separate checks to ensure Hugo will run and produce valid HTML. The Markdown linter is trying to solve a different problem.

We could definitely do more to make the style more obvious. We could explain the rules in the contributing section (this would probably also be a good check that the rules are not too complicated). We could also add advice about testing the rules locally on the command line, in an editor, using a pre-commit hook. I'm not sure if there is a way to show linting hints in the GitHub web editor.

**Reply by andrewphilipsmith** [reply]
Sorry for not getting back to you on this sooner. I think we need to be careful not to take the document-as-code metaphor too literally. Collaboratively and iteratively editing documents presents different challenges to editing code together. For now I've added a PR that updates one of the lining rules (https://github.com/alan-turing-institute/REG-handbook/pull/16).

**Comment by llewelld** [comment]
This topic came up in the discussion on #65, so I'm adding my thoughts as well.

Coming at this as someone new to the process, I find the review and automated checking to be reassuring, In other words, requiring reviews (either in the guidelines or as part of the GitHub checks) would make me more comfortable proposing a change.

I've been involved with other documentation where reviews were enforced by the pipeline. My experience was that requiring reviews but having light-touch reviewing (erring on the side of acceptance) made things work most smoothly. If I'm reading the earlier conversation right, this is pretty much in line with what @JimMadge suggested in his earlier comment https://github.com/alan-turing-institute/REG-handbook/discussions/10#discussioncomment-1948952.

So personally I'd support stating in the docs that reviews are expected.

This relates to #65 because currently the guidance seems to be a little unclear: at the top of the [Contributing](https://alan-turing-institute.github.io/REG-handbook/docs/contributing/contributing_changes/) page it says "Currently, reviews are not required", while at the bottom it says "Each pull request requires at least one approval". My preference would be to remove the note at the top, and leave the note at the bottom, but either way it should match with the agreed process.

**Reply by JimMadge** [reply]
Yes I would say this is how I was hoping the review process would work. It also captures the 'why', how a review process can (and should) be more like assistance than resistance.

It would be good to poll the team more broadly, but not everyone is already engaged with the book.

I think the real barrier to doing this is, we would need a set (more than one!) of reviewers who would be knowledgable and responsive. At the moment that is no one. In principle, I think anyone should be able to act as a reviewer, but I also think we need some 'permanent' review team to make sure PRs don't sit idle.

Volunteers are welcome!
